backend/users/views.py

```diff
@@ -367,36 +467,76 @@ def list(self, request):
         },
         tags=["User"]
     )
-    @action(detail=False, methods=['get'], url_path="self/friends")
-    def retrieve_friendlist(self, request):
+    @action(detail=False, methods=['get', 'post', 'delete'], url_path="self/friends")
+    def manage_friends(self, request, friend_name=None):
         try:
             user = request.user
             if user.is_anonymous:
                 raise PermissionError("로그인 상태가 아닙니다")
             
-            friends = Friend.objects.filter(user1_id=user)
-            if not friends.exists():
-                return Response(status=status.HTTP_204_NO_CONTENT)
-            
-            friends_data = [
-                {
-                    "username": friend.user2_id.username,
-                    "status_msg": friend.user2_id.status_msg,
-                    "status": friend.user2_id.status,
-                    "profile_img": friend.user2_id.profile_img    
-                }
-                for friend in friends
-            ]
-            return Response(friends_data, status=status.HTTP_200_OK)
+            if request.method == 'GET':    
+                friends = Friend.objects.filter(username=user)
+                if not friends.exists():
+                    return Response(status=status.HTTP_204_NO_CONTENT)
+                
+                friends_data = []
+                for friend in friends:
+                    friend_user = friend.friendname
+                    matches = Match.objects.filter((Q(match_username1=friend_user) | Q(match_username2=friend_user)) & ~Q(match_result='pending_result'))
+                    friends_data.append(
+                        {
+                            "username": friend_user.username,
+                            "status_msg": friend_user.status_msg,
+                            "status": friend_user.status,
+                            "profile_img": friend_user.profile_img,
+                            "matches": [
+                                {
+                                    'match_result': match.match_result,
+                                    'match_start_time': match.match_start_time,
+                                    'match_end_time': match.match_end_time,
+                                    'username1_grade': match.username1_grade,
+                                    'username2_grade': match.username2_grade,
+                                    'match_type': match.match_type
+                                } 
+                                for match in matches.order_by('match_end_time')[:5]
```

Comment:
for match in matches.order_by('match_end_time')[:5]
->
for match in matches.order_by('-match_end_time')[:5]